### PR TITLE
Alerting: Add state_periodic_save_batch_size config option

### PR DESCRIPTION
**Pull request by alexander-akhmetov**
**What is this feature?**

This PR adds the `state_periodic_save_batch_size` configuration option, which allows control over the batch size of alert instances inserted during periodic state saving when `alertingSaveStatePeriodic` feature flag is enabled

**Why do we need this feature?**

During periodic async state saving, a large number of alert instances in memory increase the time required to complete the sync process. Because the save happens in a single transaction, the alert instances in the database table remain locked for the entire duration, blocking alert rules operations, etc.

Adjusting the batch size can reduce the sync time, but can cause higher load on the database.

<details>
  <summary>Example</summary>

**state_periodic_save_batch_size = 1**
```
level=debug msg="Full state sync done" duration=58.325004959s instances=125100
level=debug msg="Full state sync done" duration=1m5.586675333s instances=125100
level=debug msg="Full state sync done" duration=1m9.609511708s instances=125100  
```

**state_periodic_save_batch_size = 1000**
```
level=debug msg="Full state sync done" duration=9.639794166s instances=125100
level=debug msg="Full state sync done" duration=8.174613666s instances=125100
level=debug msg="Full state sync done" duration=7.905657334s instances=125100
```

</details>

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: docs/sources/alerting/set-up/performance-limitations/index.md [hunk 1]
```diff
@@ -65,6 +65,10 @@ This can be prevented by writing to the database periodically. For this the feat
 to be enabled. By default, it saves the states every 5 minutes to the database and on each shutdown. The periodic interval
 can also be configured using the `state_periodic_save_interval` configuration flag.
 
+When periodic saving is enabled, each `state_periodic_save_interval` Grafana deletes all existing alert instances from the
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/performance-limitations/index.md:68]: not sure what the first sentence , each state periodic save interval Grafana means?

File: docs/sources/alerting/set-up/performance-limitations/index.md [hunk 2]
```diff
@@ -65,6 +65,10 @@ This can be prevented by writing to the database periodically. For this the feat
...
+database and then writes the entire current set of instances back in batches in a single transacton.
+The size of each batch can be configured using the `state_periodic_save_batch_size` configuration option.
```
Comment (brendamuir) [inline docs/sources/alerting/set-up/performance-limitations/index.md:70]:
```suggestion
Configure the size of each batch using the `state_periodic_save_batch_size` configuration option.
```

Comment (brendamuir) [reply docs/sources/alerting/set-up/performance-limitations/index.md:70]: (we try not to use passive)

File: docs/sources/alerting/set-up/performance-limitations/index.md (hunk 1 above)
Comment (alexander-akhmetov) [reply docs/sources/alerting/set-up/performance-limitations/index.md]: I updated the doc, could you have another look, please?

**Review by brendamuir — APPROVED** [review]
lgtm from docs pov

**Review by fayzal-g — APPROVED** [review]
Great stuff 👌

**Pull request merged**